### PR TITLE
docs: fix documentation for global vue3 instance handling

### LINKED ISSUE #21045: [Feature Request]: export `app` instance for vue3/vite to extend the application in preview.ts
### Is your feature request related to a problem? Please describe

Sometimes it's necessary to extend an Vue3 application with global plugins as shown in the basic example below.

```
import { createApp } from 'vue'
import MyPlugin from 'some-plugin'

import App from './App.vue'

const app = createApp(MyPlugin)
```

Such plugins can provide application-wide features on which some components may rely on.

Mounting components in Storybook in isolation will under the hood create a new vue3 instance.

[The documentation for vue3/webpack already shows how to extend the applications instance used by storybook with custom plugins](https://github.com/storybookjs/storybook/tree/next/code/frameworks/vue3-webpack5).

Also the way of vue2 using `Vue.use` is described for [Vite](https://github.com/storybookjs/storybook/tree/next/code/frameworks/vue-vite) and [Webpack](https://github.com/storybookjs/storybook/tree/next/code/frameworks/vue-webpack5).

Unfortunately the `app` instance is not exported right now, when using vite + vue3 and storybook. So the following code won't work:

```
import { app } from '@storybook/vue3-vite';
/* OR */
import { app } from '@storybook/vue3';

import MyPlugin from 'my-plugin'

app.use(MyPlugin);

export const parameters = {
  // ...
}
```

So it seems like `app` is not exported and not extensible in fact.

There seems already to be an empty doc for the documentation which has "React" as headline (I assume it was simply copied):
https://github.com/storybookjs/storybook/tree/next/code/frameworks/vue3-vite

### Describe the solution you'd like

The `app` instance used by storybook should be exported so it is extensible:

```
import { app } from '@storybook/vue3-vite';
/* OR */
import { app } from '@storybook/vue3';

import MyPlugin from 'my-plugin'

app.use(MyPlugin);

export const parameters = {
  // ...
}
```

Documentation should be added.

### Describe alternatives you've considered

_No response_

### Are you able to assist to bring the feature to reality?

yes, I can

### Additional context

My package.json when testing this looked like this:

```
  "dependencies": {
    "sass": "1.58.0",
    "vue": "^3.2.45"
  },
  "devDependencies": {
    "@storybook/addon-essentials": "^7.0.0-beta.45",
    "@storybook/addon-interactions": "^7.0.0-beta.45",
    "@storybook/addon-links": "^7.0.0-beta.45",
    "@storybook/blocks": "^7.0.0-beta.45",
    "@storybook/testing-library": "^0.0.14-next.1",
    "@storybook/vue3": "^7.0.0-beta.45",
    "@storybook/vue3-vite": "^7.0.0-beta.45",
    "@vitejs/plugin-vue": "^4.0.0",
    "react": "^18.2.0",
    "react-dom": "^18.2.0",
    "storybook": "^7.0.0-beta.45",
    "typescript": "^4.9.3",
    "vite": "^4.1.0",
    "vue-tsc": "^1.0.24"
  }
```

### PR DESCRIPTION
**Pull request by d-koppenhagen**
Closes #21045

## Checklist

- [x] Make sure your changes are tested (stories and/or unit, integration, or end-to-end tests)
- [x] Make sure to add/update documentation regarding your changes
- [x] If you are deprecating/removing a feature, make sure to update
      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

#### Maintainers

- [ ] If this PR should be tested against many or all sandboxes,
      make sure to add the `ci:merged` or `ci:daily` GH label to it.
- [ ] Make sure this PR contains **one** of the labels below.

`["cleanup", "BREAKING CHANGE", "feature request", "bug", "documentation", "maintenance", "dependencies", "other"]`

### CODE REVIEW

File: code/frameworks/vue3-vite/README.md [hunk 1]
```diff
@@ -1 +1,45 @@
...
+Storybook also comes with a lot of [addons](https://storybook.js.org/addons) and a great API to customize as you wish.
+You can also build a [static version](https://storybook.js.org/docs/vue/sharing/publish-storybook) of your Storybook and deploy it anywhere you want.
+
+## Extending the Vue application
+
+Storybook creates a [Vue 3 application](https://v3.vuejs.org/api/application-api.html#application-api) for your component preview.
+When using global custom components (`app.component`), directives (`app.directive`), extensions (`app.use`), or other application methods, you will need to configure those in the `./storybook/preview.js` file.
+
+Therefore, Storybook provides you a a `setup` function from `@storybook/vue3`.
+This function receives a callback with the app instance of Storybooks Vue app passed that you can interact with.
```
Comment (jonniebigodes) [inline code/frameworks/vue3-vite/README.md:31]: @d-koppenhagen, it seems that this paragraph needs a bit of adjustment. When you're able, can you update it to the following for a bit more clarity:
"Therefore, Storybook provides you with a `setup` function exported from this package, which receives as a callback your Storybook instance, which you can interact with and add your custom configuration."

File: code/frameworks/vue3-webpack5/README.md [hunk 2]
```diff
@@ -24,20 +24,22 @@ You can also build a [static version](https://storybook.js.org/docs/vue/sharing/
...
 ## Extending the Vue application
 
-Storybook creates a [Vue 3 application](https://v3.vuejs.org/api/application-api.html#application-api) for your component preview which can be imported as `import { app } from '@storybook/vue3'`.
-
+Storybook creates a [Vue 3 application](https://v3.vuejs.org/api/application-api.html#application-api) for your component preview.
 When using global custom components (`app.component`), directives (`app.directive`), extensions (`app.use`), or other application methods, you will need to configure those in the `./storybook/preview.js` file.
 
-For example:
+Therefore, Storybook provides you a a `setup` function from `@storybook/vue3`.
+This function receives a callback with the app instance of Storybooks Vue app passed that you can interact with.
```
Comment (jonniebigodes) [inline code/frameworks/vue3-webpack5/README.md:31]: @d-koppenhagen, can you make the same adjustment as previously mentioned for the vue3-vite Readme when you're able?

**Review by jonniebigodes — CHANGES_REQUESTED** [review]
@d-koppenhagen, thanks for the time and effort you took to set up this pull request and help us improve our documentation. We appreciate it 🙏 ! I left a couple of items for you to look at when you're able so that we can merge this.

Let me know once you've addressed them, and we'll go from there.

Looking forward to hearing from you. I hope you have a fantastic weekend. 

Stay safe

File: code/frameworks/vue3-vite/README.md (hunk 1 above)
Comment (d-koppenhagen) [reply code/frameworks/vue3-vite/README.md]: Okay, I replaced the sentence with your suggestion.

**Comment by d-koppenhagen** [comment]
@jonniebigodes thanks for your feedback. I added your suggestions. Please let me know if there is anything else I can do. Best wishes, Danny.

**Comment by jonniebigodes** [comment]
@d-koppenhagen, thanks for addressing the feedback so promptly. Appreciate it 🙏 ! I've checked, and this looks good on my end. Thanks once again for taking the time to update the documentation. We appreciate it. I'll merge this in a bit.

Hope you have a great week.

Stay safe

**Review by jonniebigodes — APPROVED** [review]

**Pull request merged**